lib/cmock_generator.rb

```diff
@@ -206,6 +207,9 @@ def create_mock_implementation(file, function)
     args_string += (", " + function[:var_arg]) unless (function[:var_arg].nil?)
 
     # Create mock function
+    if (@weak)
+        file << "#{function_mod_and_rettype} #{function[:name]}(#{args_string}) __attribute ((weak));\n"
```

Comment:
ok, i extended the configurability of this weak symbols option. For IAR i based myself on http://stackoverflow.com/questions/25619377/what-is-the-iar-equivalent-of-attribute-weak-alias-f  No way of testing the IAR case for me however.